### PR TITLE
add no_cast to _mtchsel

### PR DESCRIPTION
**Pull request by Tianhao-Gu**
example generated genome_attrbs.jsonl  file.

```
{"kbase_id": "69037_2_1", "Contamination": 0.85, "Completeness": 88.28, "_key": "6c95a1d46e809f84063dccf93700ffa9", "coll": "GROW", "load_ver": "tian_test_taskfarmer", "_mtchsel": [], "user_genome": "69037_2_1_kbase", "classification": "d__Bacteria;p__Verrucomicrobiota;c__Verrucomicrobiae;o__Chthoniobacterales;f__Chthoniobacteraceae;g__Chth-196;s__", "fastani_reference": null, "fastani_reference_radius": null, "fastani_taxonomy": null, "fastani_ani": null, "fastani_af": null, "closest_placement_reference": "GCA_945883955.1", "closest_placement_radius": 95.0, "closest_placement_taxonomy": "d__Bacteria;p__Verrucomicrobiota;c__Verrucomicrobiae;o__Chthoniobacterales;f__Chthoniobacteraceae;g__Chth-196;s__Chth-196 sp945883955", "closest_placement_ani": 76.9, "closest_placement_af": 0.093, "pplacer_taxonomy": "d__Bacteria;p__Verrucomicrobiota;c__Verrucomicrobiae;o__Chthoniobacterales;f__Chthoniobacteraceae;g__Chth-196;s__", "classification_method": "taxonomic classification defined by topology and ANI", "note": "classification based on placement in class-level tree", "other_related_references(genome_id,species_name,radius,ANI,AF)": "GCA_945874745.1, s__Chth-196 sp945874745, 95.0, 76.48, 0.079; GCA_903916085.1, s__Chth-196 sp903916085, 95.0, 76.36, 0.047; GCA_003402745.1, s__Chth-196 sp003402745, 95.0, 76.19, 0.062; GCA_945906945.1, s__Chth-196 sp945906945, 95.0, 75.8, 0.052; GCA_903919125.1, s__Chth-196 sp903919125, 95.0, 75.58, 0.036", "msa_percent": 81.85, "translation_table": 11, "red_value": "0.93961", "warnings": null, "kbase_display_name": "altamaha_2019_sw_WHONDRS-S19S_0010_A_bin_34_mag_assembly", "_upas": {"KBaseGenomeAnnotations.Assembly": "69037/2/1", "KBaseGenomes.Genome": "69037/3/1"}, "kbase_genome_size": 3670106, "kbase_gc_content": 0.5997, "kbase_num_contigs": 233, "kbase_num_cds": 3053, "kbase_num_protein_encoding_genes": 3053, "kbase_sample_id": "79db2b9e-0581-4a53-9154-5aa6030aed02"}
```

### CODE REVIEW

File: src/loaders/common/loader_helper.py [hunk 1]
```diff
@@ -114,8 +115,12 @@ def process_columnar_meta(
     spec = load_spec(product_id, kbase_collection)
     columns = list()
     for col_spec in spec.columns:
-        key, col_type = col_spec.key, col_spec.type
-        values = _convert_values_to_type(docs, key, col_type, ignore_missing=ignore_missing)
+        key, col_type, no_cast = col_spec.key, col_spec.type, col_spec.no_cast
```
Comment (MrCreosote) [inline src/loaders/common/loader_helper.py:118]: I would just access these from col_type personally

Comment (Tianhao-Gu) [reply src/loaders/common/loader_helper.py:118]: you mean `col_spec`? like pass `col_spec` to the `_convert_values_to_type`?

Comment (MrCreosote) [reply src/loaders/common/loader_helper.py:118]: No, just get rid of that line and do 
```python
        values = _convert_values_to_type(docs,
                                         col_spec.key,
                                         col_spec.col_type,
                                         ignore_missing=ignore_missing,
                                         no_cast=col_spec.no_cast)
```

and same for the `col_type` references below

**Comment by MrCreosote** [comment]
Ugh, we have the same problem for `kbase_ids`, which is visible...

Maybe this needs a rethink. Maybe have a container field like the type field, but for now which only has a type of ARRAY?

File: src/loaders/common/loader_helper.py (hunk 1 above)
Comment (Tianhao-Gu) [reply src/loaders/common/loader_helper.py]: 👍

**Comment by Tianhao-Gu** [comment]
> Ugh, we have the same problem for `kbase_ids`, which is visible...
> 
> Maybe this needs a rethink. Maybe have a container field like the type field, but for now which only has a type of ARRAY?

kbase_ids is not visible.

**Comment by MrCreosote** [comment]
Oh good, I forgot we decided to hide that for now. Ok, let's keep things as is for now then

**Review by MrCreosote — APPROVED** [review]
LGTM

**Pull request merged**